src/chainstate/stacks/boot/pox_2_tests.rs

```diff
@@ -483,3 +483,324 @@ fn test_simple_pox_lockup_transition_pox_2() {
+                } else {
+                    // v2 reward cycles have begun, so reward addrs should be read from PoX2 which is Bob + Alice
+                    assert_eq!(reward_addrs.len(), 2);
+                    assert_eq!(
+                        (reward_addrs[0].0).version,
+                        AddressHashMode::SerializeP2PKH.to_version_testnet()
+                    );
+                    assert_eq!((reward_addrs[0].0).bytes, key_to_stacks_addr(&bob).bytes);
+                    assert_eq!(reward_addrs[0].1, 512 * POX_THRESHOLD_STEPS_USTX);
+
+                    assert_eq!(
+                        (reward_addrs[1].0).version,
+                        AddressHashMode::SerializeP2PKH.to_version_testnet()
+                    );
+                    assert_eq!((reward_addrs[1].0).bytes, key_to_stacks_addr(&alice).bytes);
+                    assert_eq!(reward_addrs[1].1, 1024 * POX_THRESHOLD_STEPS_USTX);
+                }
+            } else {
+                // no reward addresses
+                assert_eq!(reward_addrs.len(), 0);
+            }
+        }
+    }
+
+    // now let's check some tx receipts
+
+    let alice_address = key_to_stacks_addr(&alice);
+    let bob_address = key_to_stacks_addr(&bob);
+    let blocks = observer.get_blocks();
+
+    let mut alice_txs = HashMap::new();
+    let mut bob_txs = HashMap::new();
+
+    eprintln!("Alice addr: {}", alice_address);
+    eprintln!("Bob addr: {}", bob_address);
+
+    for b in blocks.into_iter() {
+        for r in b.receipts.into_iter() {
+            if let TransactionOrigin::Stacks(ref t) = r.transaction {
+                let addr = t.auth.origin().address_testnet();
+                eprintln!("TX addr: {}", addr);
+                if addr == alice_address {
+                    alice_txs.insert(t.auth.get_origin_nonce(), r);
+                } else if addr == bob_address {
+                    bob_txs.insert(t.auth.get_origin_nonce(), r);
+                }
+            }
+        }
+    }
+
+    assert_eq!(alice_txs.len(), 3, "Alice should have 3 confirmed txs");
+    assert_eq!(bob_txs.len(), 1, "Bob should have 1 confirmed txs");
+
+    assert!(
+        match alice_txs.get(&0).unwrap().result {
+            Value::Response(ref r) => r.committed,
+            _ => false,
+        },
+        "Alice tx0 should have committed okay"
+    );
+
+    assert!(
+        match alice_txs.get(&1).unwrap().result {
+            Value::Response(ref r) => r.committed,
+            _ => false,
+        },
+        "Alice tx1 should have committed okay"
+    );
+
+    assert_eq!(
+        alice_txs.get(&2).unwrap().result,
+        Value::err_none(),
+        "Alice tx2 should have resulted in a runtime error"
+    );
+
+    assert!(
+        match bob_txs.get(&0).unwrap().result {
+            Value::Response(ref r) => r.committed,
+            _ => false,
+        },
+        "Bob tx0 should have committed okay"
+    );
+}
```

Comment:
Do you think there could be a corresponding test for `delegate-stack-extend` to verify that Alice can delegate some STX to Bob, and then extend the delegation across the v1 -> v2 upgrade?

Reply:
Yes, I will add

Reply:
Added in 20f156fe9c9434ba6c98d61f1c51ab6a96ed8dcc